### PR TITLE
ZER-163-ProductPrice-UTests

### PR DESCRIPTION
**Pull request by PrincTwilig**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

Add different and meaningful at least 5 tests to PriceCategory model.

## Summary of changes

Result of Rspec test

image

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by AndriiStegnytskyi — APPROVED** [review]

File: spec/models/product_price_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+LOCAL_PREFIX_PRODUCT_PRICE = "activerecord.errors.models.product_price.attributes"
```
Comment (PavloMS) [inline spec/models/product_price_spec.rb:4]: Why do you need this?

File: spec/models/product_price_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+RSpec.describe ProductPrice, type: :model do
+  subject { build(:product_price) }
+  describe "validations" do
+    describe "price" do
```
Comment (PavloMS) [inline spec/models/product_price_spec.rb:9]: context

File: spec/models/product_price_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      it {
+        should validate_presence_of(:price)
+      }
+      it {
+        expect(subject.price).to be_a(Float)
+      }
+    end
+    describe "category" do
```
Comment (PavloMS) [inline spec/models/product_price_spec.rb:17]: context

File: spec/models/product_price_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      it {
+        should validate_presence_of(:category)
+      }
+      it {
+        should validate_inclusion_of(:category).in?([0, 1, 2])
+      }
+      describe "enum_names" do
```
Comment (PavloMS) [inline spec/models/product_price_spec.rb:24]: remove this describe from describe "validations" do

File: spec/models/product_price_spec.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        it {
+          subject.category = 0
```
Comment (PavloMS) [inline spec/models/product_price_spec.rb:26]: use let to change category

File: spec/models/product_price_spec.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+          expect(subject.category).to eq("LOW")
+        }
+        it {
+          subject.category = 1
+          expect(subject.category).to eq("MEDIUM")
+        }
+        it {
+          subject.category = 2
```
Comment (PavloMS) [inline spec/models/product_price_spec.rb:34]: same as above ^^

File: spec/models/product_price_spec.rb (hunk 1 above)
Comment (PrincTwilig) [reply spec/models/product_price_spec.rb:4]: removed

File: spec/models/product_price_spec.rb (hunk 5 above)
Comment (PrincTwilig) [reply spec/models/product_price_spec.rb:26]: let cant be used in "it" block

image

if let is out of "it" block, that dousnt change category number

image
(by default category is "MEDIUM")

**Review by KostiaFed — APPROVED** [review]

**Review by AndriiStegnytskyi — APPROVED** [review]

**Review by PavloMS — APPROVED** [review]

**Pull request closed without merging**